### PR TITLE
Fix ImagenetDatasetHparams bug

### LINKED ISSUE #1111: Bug in ImagenetDatasetHparams with validation set
composer/datasets/imagenet.py
line 171
transforms.Resize(self.resize_size),

Should check if self.resize_size>0.
The easiest fix would be to do the same as in line 238.

otherwise is going to raise error:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 530, in __next__
    data = self._next_data()
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 1224, in _next_data
    return self._process_data(data)
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 1250, in _process_data
    data.reraise()
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torch/_utils.py", line 457, in reraise
    raise exception
ValueError: Caught ValueError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torch/utils/data/_utils/worker.py", line 287, in _worker_loop
    data = fetcher.fetch(index)
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torch/utils/data/_utils/fetch.py", line 49, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torch/utils/data/_utils/fetch.py", line 49, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torchvision/datasets/folder.py", line 232, in __getitem__
    sample = self.transform(sample)
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torchvision/transforms/transforms.py", line 95, in __call__
    img = t(img)
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torchvision/transforms/transforms.py", line 349, in forward
    return F.resize(img, self.size, self.interpolation, self.max_size, self.antialias)
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torchvision/transforms/functional.py", line 436, in resize
    return F_pil.resize(img, size=size, interpolation=pil_interpolation, max_size=max_size)
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/torchvision/transforms/functional_pil.py", line 258, in resize
    return img.resize((new_w, new_h), interpolation)
  File "/opt/conda/envs/spartns/lib/python3.10/site-packages/PIL/Image.py", line 2062, in resize
    return self._new(self.im.resize(size, resample, box))
ValueError: height and width must be > 0

### PR DESCRIPTION
**Pull request by nik-mosaic**
Closes #1111 and similar bug in ade20k_hparams.

### CODE REVIEW

File: composer/datasets/imagenet_hparams.py [hunk 1]
```diff
@@ -182,10 +182,12 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataLoaderHpara
...
                 split = 'train'
             else:
-                transformation = transforms.Compose([
-                    transforms.Resize(self.resize_size),
-                    transforms.CenterCrop(self.crop_size),
-                ])
+                val_resize_size = self.resize_size
+                val_transforms: List[torch.nn.Module] = []
+                if val_resize_size > 0:
+                    val_transforms.append(transforms.Resize(val_resize_size))
```
Comment (mvpatel2000) [inline composer/datasets/imagenet_hparams.py:188]:
```suggestion
                val_transforms: List[torch.nn.Module] = []
                if self.resize_size > 0:
                    val_transforms.append(transforms.Resize(self.resize_size))
```

File: composer/datasets/imagenet_hparams.py [hunk 2]
```diff
@@ -182,10 +182,12 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataLoaderHpara
...
+                val_transforms += [transforms.CenterCrop(self.crop_size)]
```
Comment (mvpatel2000) [inline composer/datasets/imagenet_hparams.py:189]: nit: prefer append
```suggestion
                val_transforms.append(transforms.CenterCrop(self.crop_size))
```

**Review by mvpatel2000 — APPROVED** [review]
LGTM

File: composer/datasets/imagenet_hparams.py (hunk 1 above)
Comment (nik-mosaic) [reply composer/datasets/imagenet_hparams.py]: Changed, and changed similar code in `if self.is_train` branch.

**Comment by dskhudia** [comment]
testcase?

**Comment by nik-mosaic** [comment]
> testcase? Is it already covered by existing tests?

There don't appear to be any tests for most of the ImagenetDatasetHparams class. One test exists in test_trainer.py using the class, but it is primarily testing FFCV datasets, so it sets `use_ffcv=True`. The ImagenetDatasetHparams class supports three cases: FFCV, a synthetic dataset, and the `else` case, which loads in a dataset from a local directory and applies the above transformations. So this code can be tested but would require a local image dataset.

**Review by hanlint — APPROVED** [review]
test case is probably less needed here, LGTM! Can you resolve the jenkins error and merge? May need to a dummy commit to force a re-test.

**Review by knighton — APPROVED** [review]
LGTM

Noting that the parallel imagenet.py transforms look ok

**Pull request merged**